domain/consensus/utils/txscript/engine.go

```diff
@@ -358,9 +361,9 @@ func (vm *Engine) currentScript() []parsedOpcode {
 
 // checkHashTypeEncoding returns whether or not the passed hashtype adheres to
 // the strict encoding requirements if enabled.
-func (vm *Engine) checkHashTypeEncoding(hashType SigHashType) error {
-	sigHashType := hashType & ^SigHashAnyOneCanPay
-	if sigHashType < SigHashAll || sigHashType > SigHashSingle {
+func (vm *Engine) checkHashTypeEncoding(hashType consensushashing.SigHashType) error {
+	sigHashType := hashType & ^consensushashing.SigHashAnyOneCanPay
```

Comment:
(FYI &^ is a single operation in golang, but I think I like the space between them (altough `^` by itself is xor))